Quickstart refinements (#13244)
PrefectHQ/prefect: docs/getting-started/quickstart.md

# Quickstart

> Prefect is a workflow orchestration platform for building data pipelines, background jobs, LLM Chains, and more. With Prefect, you can elevate a Python script into a scheduled, observable, resilient, remotely execuateble workflow in a few minutes. Let's get started!

Reviewer: Small typo in executable- sorry on mobile so I can't make a suggested change.

> Prefect is a workflow orchestration platform for building data pipelines, background jobs, LLM Chains, and more. With Prefect, you can elevate a Python script into a scheduled, observable, resilient, remotely execuateble workflow in a few minutes. Let's get started!

Reviewer 2: ```suggestion
Prefect is a workflow orchestration platform for building data pipelines, background jobs, LLM Chains, and more. With Prefect, you can elevate a Python script into a scheduled, observable, resilient, remotely executable workflow in a few minutes. Let's get started!
```

## Setup

Here's a basic script that fetches statistics about the [main Prefect GitHub repository](https://github.com/PrefectHQ/prefect).

```python title="my_gh_workflow.py"
import httpx   # an HTTP client library

def get_repo_info(repo_owner: str, repo_name: str):
    """Get info about a repo"""
    url = f"https://api.github.com/repos/{repo_owner}/{repo_name}"
    api_response = httpx.get(url)
    api_response.raise_for_status()
    repo_info = api_response.json()
    return repo_info

def get_contributors(repo_info: dict):
    """Get contributors for a repo"""
    contributors_url = repo_info["contributors_url"]
    response = httpx.get(contributors_url)
    response.raise_for_status()
    contributors = response.json()
    return contributors

def repo_info(repo_owner: str = "PrefectHQ", repo_name: str = "prefect"):
    """Given a repo, logs the number of stargazers and contributors"""
    repo_info = get_repo_info(repo_owner, repo_name)
    print(f"Stars 🌠 : {repo_info['stargazers_count']}")

    contributors = get_contributors(repo_info)
    print(f"Number of contributors 👷: {len(contributors)}")

if __name__ == "__main__":
    repo_info()
```

> Copy this script into a file named `my_gh_workflow.py`. Run it locally to ensure your Python environment works. Let's make this script schedulable, observable, resilient, and capable of running anywhere.

Reviewer: Is the "let's make this" section still meant to be in here? It feels a little disjointed.

Reviewer 2: I think it's meant to be a transition from "you just have a script" to "look what Prefect can do for you".  Is that transition improved by the suggested change added below?

> Copy this script into a file named `my_gh_workflow.py`. Run it locally to ensure your Python environment works. Let's make this script schedulable, observable, resilient, and capable of running anywhere.

Reviewer 2: ```suggestion
Copy this script into a file named `my_gh_workflow.py`. Run it locally to ensure your Python environment works.

 Now let's use Prefect to make this script schedulable, observable, resilient, and capable of running anywhere.
```

## Step 1: Install Prefect

```bash
pip install -U prefect
```

See the [install guide](/getting-started/installation/) for more detailed installation instructions.

## Step 2: Connect to Prefect's API

> Much of Prefect's functionality is backed by an API. You can [host Prefect server](/guides/host/) yourself if you'd like, but the easiest way to get started is to use the API hosted by Prefect Cloud:

Reviewer: Does the mention of server here increase complexity?

> Much of Prefect's functionality is backed by an API. You can [host Prefect server](/guides/host/) yourself if you'd like, but the easiest way to get started is to use the API hosted by Prefect Cloud:

Reviewer 2: ```suggestion
Much of Prefect's functionality is backed by an API. You can [host a Prefect server instance](/guides/host/) yourself if you'd like, but the easiest way to get started is to use the API hosted by Prefect Cloud:
```

## Step 4: Choose a remote infrastructure location

Let's get this workflow running on infrastructure other than your local machine!

> Prefect refers to infrastructure that can run worflows as [work pools](/concepts/work-pools/). 

Reviewer 2: ```suggestion
Prefect refers to infrastructure that can run workflows as a [work pool](/concepts/work-pools/). 
```

> Prefect can orchestrate workflows that run remotely on many [types of infrastructure](/work-pools/#work-pool-types), but here we'll use a [Prefect Managed work pool](/guides/managed-execution/) so that Prefect can run this flow for us.

Reviewer 2: ```suggestion
Prefect can orchestrate workflows that run remotely on many [types of infrastructure](/work-pools/#work-pool-types).  We'll use a [Prefect Managed work pool](/guides/managed-execution/) so that Prefect can run this flow for us.
```

You should see a message in the CLI that your work pool was created.

> Feel free to check out your new work pool on the **Work Pools** page in the UI.

Reviewer: Is this line helpful? I'd vote to remove it as it pulls the user away from the main focus.

Reviewer 2: Good question. I think we want to get the user checking out the UI at some point, but maybe not until they run the deployment.

Reviewer 2: You could do it all in Python code + CLI. The UI is a core part of the product that I think we want folks interacting with. That said, I hear you on bouncing back and forth being distracting.

>         entrypoint="my_gh_workflow.py:repo_info",

Reviewer 2: Making the source a Prefect repo would be good.

    ).deploy(
        name="my-first-deployment",

>         work_pool_name="my-managed-pool",

Reviewer: Actually a content from line 186 - I suggest we point out more clearly that here they're using code we've stored in GitHub.
